### PR TITLE
made sealed class for AddRatingScreen

### PR DESCRIPTION
**Pull request by Maniii97**

### CODE REVIEW

**Comment by basakjeet08** [comment]
Issues :- 

- Instead of using Events/Actions to pass data from View Model to the UI use MutableStateFlow and observe the data.Pass the data directly to the  UI composable function parameters.

File: app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/navigation/AddRatingNavGraph.kt [hunk 1]
```diff
@@ -43,7 +44,11 @@ fun AddRatingNavGraph(
                 content = {
                     AddRatingScreen(
                         navController = navController,
-                        myViewModel = myViewModel
+                        action = myViewModel::action,
+                        teacherName = teacherData.name,
+                        markingRating = myViewModel._userInputReview.collectAsState().value.markingRating,
+                        attendanceRating = myViewModel._userInputReview.collectAsState().value.attendanceRating,
+                        teachingRating = myViewModel._userInputReview.collectAsState().value.teachingRating
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/navigation/AddRatingNavGraph.kt:51]: Extract out the state in a variable and use it in the following fields

File: app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/navigation/AddRatingNavGraph.kt [hunk 2]
```diff
@@ -53,8 +58,14 @@ fun AddRatingNavGraph(
...
-                        myViewModel = myViewModel,
-                        refreshTeacherReviews = refreshTeacherReviews
+                        refreshTeacherReviews = refreshTeacherReviews,
+                        action = myViewModel::action,
+                        addReviewApiState = myViewModel.addReviewApiState,
+                        teacherName = teacherData.name,
+                        overallReview = myViewModel._userInputReview.collectAsState().value.overallReview,
+                        markingReview = myViewModel._userInputReview.collectAsState().value.markingReview,
+                        attendanceReview = myViewModel._userInputReview.collectAsState().value.attendanceReview,
+                        teachingReview = myViewModel._userInputReview.collectAsState().value.teachingReview
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/navigation/AddRatingNavGraph.kt:68]: Extract out the state in a variable and use it in the following fields

File: app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/stateholder/AddReviewViewModel.kt [hunk 3]
```diff
@@ -21,26 +25,10 @@ class AddReviewViewModel @Inject constructor(
...
-
-    var userInputAttendanceReview: String by mutableStateOf("")
-        private set
-
-    var userInputTeachingReview: String by mutableStateOf("")
-        private set
+    var _userInputReview = MutableStateFlow(
+        ReviewStateFlow()
+    )
+    val userInputReview = _userInputReview.asStateFlow()
```
Comment (dead8309) [inline app/src/main/java/in/iot/lab/teacherreview/feature_teacherlist/presentation/stateholder/AddReviewViewModel.kt:31]: use this pattern for exposing states from vm to screens

```kt
// ViewModel
class VM : ViewModel() {
private val _state= MutableStateFlow(YOUR_STATE_CLASS())
val state = _state.asStateFlow()
}

// Screen
@Composable
fun Screen(viewModel: VM) {
 val state by viewModel.state.collectAsState()
}
```
You can use any name for state variable

**Pull request merged**